### ISSUE TITLE
Format failure with combination of space preceeding comment and `with rollup`

### ISSUE THREAD

**Issue by rustprooflabs** [sqlfluff/sqlfluff#6524]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I am attempting to format a SQL file intended for MS SQL Server (t-sql) with this type of command:

```bash
sqlfluff format "test_file.sql"
```

The result is a failure with this output.

```bash
WARNING    Fixes for LT02 not applied, as it would result in an unparsable file. Please report this as a bug with a minimal query which demonstrates this warning.
WARNING    One fix for LT13 not applied, it would re-cause the same error.
WARNING    One fix for LT13 not applied, it would re-cause the same error.
== [test_file.sql] FAIL
L:   1 | P:   1 | LT02 | First line should not be indented.
                       | [layout.indent]
L:   1 | P:   1 | LT13 | Files must not begin with newlines or whitespace.
                       | [layout.start_of_file]
  [1 templating/parsing errors found]
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [2 unfixable linting violations found]
```

### Expected Behaviour

I expected it to successfully format the SQL file.

### Observed Behaviour

Failed with message indicated above.  I can avoid the error by one of two methods described in How to reproduce, though should not have to do those manual cleanups to avoid this error.

### How to reproduce

The SQL that fails for me is this 2 line example.  Note the space preceding the comment on line 1.

```sql
 -- checks 
SELECT st, count(*), count(DISTINCT id) FROM #3 GROUP BY st with rollup;
```

This works: remove the preceding space on line 1.

```sql
-- checks 
SELECT st, count(*), count(DISTINCT id) FROM #3 GROUP BY st with rollup;
```

This also works: remove `with rollup` (while still leaving the preceding space on line 1).

```sql
 -- checks 
SELECT st, count(*), count(DISTINCT id) FROM #3 GROUP BY st;
```

### Dialect

tsql

### Version

```bash
python --version
Python 3.12.4

sqlfluff --version
sqlfluff, version 3.1.0
```

### Configuration

```ini
# See https://docs.sqlfluff.com/en/stable/configuration.html#default-configuration

[sqlfluff]
dialect = tsql
max_line_length = 80
#templater = placeholder # e.g. %(term)s
templater = python
# Use all processors but 1.  Use positive number (e.g. 4) to use that many.
processes = -1
# Allows lists of columns to share a line if under max_line_length
exclude_rules = layout.select_targets

# Very large files can make the parser effectively hang.
# Set to 0 to disable.  Default was 20000 at time of writing.
large_file_skip_byte_limit = 0


[sqlfluff:templater:python:context]
# Parameters used in SQL code need added here to parse
# Specific values don't matter too much, SQLFluff is only concerned with formatting
term = '202430'
dynamic_filter = ' something = 1'
slice_column = 'race'

[sqlfluff:indentation]
indented_joins = True
allow_implicit_indents = True

[sqlfluff:rules:capitalisation.keywords]
capitalisation_policy = upper

[sqlfluff:layout:type:comma]
line_position = leading

[sqlfluff:layout:type:where_clause]
# This should force clause on dedicated line -- doesn't seem to though
# https://github.com/sqlfluff/sqlfluff/issues/5909#issuecomment-2171770323
line_position = alone:strict

[sqlfluff:layout:type:from_clause]
# This should force clause on dedicated line -- doesn't seem to though
# https://github.com/sqlfluff/sqlfluff/issues/5909#issuecomment-2171770323
line_position = alone:strict

[sqlfluff:rules:capitalisation.identifiers]
# Disable for tables/views. Couldn't find something to work with dbo.vTerm.
unquoted_identifiers_policy = all

# Options: 'consistent', 'upper', 'lower', 'pascal', 'capitalise', 'snake', 'camel'
#    Per: https://docs.sqlfluff.com/en/stable/rules.html
extended_capitalisation_policy = lower

[sqlfluff:layout:type:statement_terminator]
# This should force clause on dedicated line -- doesn't seem to though
# https://github.com/sqlfluff/sqlfluff/issues/5909#issuecomment-2171770323
line_position = alone:strict
```

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
